rosbag2/src/rosbag2/rosbag2.cpp

```diff
@@ -48,12 +51,7 @@ void Rosbag2::record(const std::string & file_name, const std::vector<std::strin
     throw std::runtime_error("No storage could be initialized. Abort");
   }
 
-  auto node = std::make_shared<Rosbag2Node>("rosbag2");
-
-  auto topics_and_types = topic_names.empty() ?
-    node->get_all_topics_with_types() :
-    node->get_topics_with_types(topic_names);
-
+  auto topics_and_types = node_->get_topics_with_types(topic_names);
   if (topics_and_types.empty()) {
```

Comment:
I think this check should go into the other record function

Reply:
As long as we do not have topic discovery after startup we should also report if we could not detect the topics that were explicitely specified. So I think is is an error for both `record` overloads and the location is fine.